File: 0081-search-in-rotated-sorted-array-ii/0081-search-in-rotated-sorted-array-ii.cpp
class Solution {
public:
    bool search(vector<int>& v, int t) {
        int n = v.size();
        int s = 0, e = n - 1, mid = 0;
        mid = s + (e - s) / 2;

        while (s <= e) {
            mid = s + (e - s) / 2;
            if (v[mid] == t)
                return true;
            if (v[s] == v[mid] && v[mid] == v[e]) {
                s = s + 1;
                e = e - 1;
                continue;
            }

            else if(v[s] <= v[mid] ){
                if(v[s] <= t && v[mid] > t)
                    e = mid - 1;
                else
                    s = mid + 1;
            }
            else {
                if(v[mid]<t && v[e]>=t)
                    s = mid +1;
                else
                    e = mid -1;
            }
        }
        return false;
    }
};

File: 0081-search-in-rotated-sorted-array-ii/0081-search-in-rotated-sorted-array-ii.py
class Solution:
    def search(self, nums: List[int], t: int) -> bool:
        n = len(nums)
        s = 0
        e = n-1
        mid = s - (s - e)//2

        while(s <= e):
            mid = s - (s - e)//2
            if(nums[mid] == t):
                return True
            if(nums[s] == nums[mid] and nums[mid] == nums[e]):
                s= s+1
                e= e-1

            elif(nums[s] <= nums[mid]):
                if(nums[s] <= t and nums[mid] > t):
                    e = mid-1
                else:
                    s = mid+1
            else:
                if(nums[mid] < t and nums[e]>= t):
                    s = mid+1
                else:
                    e = mid - 1

        return False